code/projects/app/src/app/shared/image-comparison/image-comparison.service.ts

```diff
@@ -0,0 +1,49 @@
+import { computed, Injectable, signal } from "@angular/core";
+
+@Injectable({
+  providedIn: "root",
+})
+export class ImageComparisonService {
+  public left = signal<ImageData | undefined>(undefined);
+  public right = signal<ImageData | undefined>(undefined);
+
+  public images = computed(() => {
+    return [this.left(), this.right()];
+  });
+
+  public addImage(which: "left" | "right", image: ImageData): Promise<boolean> {
+    // Resolve false only when the image is not the same size as the other image
+    const promise = new Promise<boolean>((resolve) => {
+      const left = this.left();
+      const right = this.right();
+      let canAdd = false;
+
+      switch (which) {
+        case "left":
+          canAdd =
+            right == null ||
+            (right.width == image.width && right.height == image.height);
+          if (canAdd) this.left.set(image);
+          break;
+
+        case "right":
+          canAdd =
+            left == null ||
+            (left.width == image.width && left.height == image.height);
+          if (canAdd) this.right.set(image);
+          break;
```

Comment:
https://github.com/PWR-ACS-SE-24/EoSaKAfDaD/pull/34#discussion_r1873328267

Reply:
Right, I didn't notice that. It can still be shortened if instead of `const values = { ... }`, we'd use
```TS
const oppositeImageValue = { left: this.right(), right: this.left() };
```

Of course the current version is also fine.

Reply:
Sure, I'll look into that.